.github/workflows/ci_cd.yml

```diff
@@ -144,8 +144,17 @@ jobs:
         run: |
           echo "BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
 
+      - name: Verify conditions for building and pushing the image
+        run: |
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            echo "IS_A_RELEASE=true" >> $GITHUB_ENV
+          else
+            echo "IS_A_RELEASE=false" >> $GITHUB_ENV
+          fi
+
       - name: Build image and export to GHCR
         uses: docker/build-push-action@v5
+        if: env.IS_A_RELEASE == 'false'
         with:
           context: .
           file: ./docker/Dockerfile
```

Comment:
The line right below this (which I can't comment on..) is ``push: ${{ github.ref == 'refs/heads/main' }}``. Is that still correct, or does it need a check for tags?

Reply:
Aha, and the release tag is only pushed below, correct?

Reply:
Yep that's right